### PR TITLE
Use Backendless Request

### PR DESCRIPTION
**Pull request by illyaV**
- use backendless-request for all async API methods
- add a warning for all sync methods
- refactoring of sync-request
- remove caching 
- remove UI state
- add node.js examples for files

### CODE REVIEW

File: src/file/files.js [hunk 1]
```diff
@@ -6,244 +6,155 @@ import Private from '../private'
...
 
-  xhr.open(options.method, options.url, !!async)
+const sanitizeFileName = fileName => encodeURIComponent(fileName).replace(/'/g, '%27').replace(/"/g, '%22')
 
-  if (encoded) {
-    xhr.setRequestHeader('Content-Type', 'text/plain')
-  } else {
-    xhr.setRequestHeader('content-type', 'multipart/form-data; boundary=' + boundary)
-  }
+const toBiteArray = content => {
```
Comment (Valodya) [inline src/file/files.js]: `toByteArray`

File: src/file/files.js [hunk 2]
```diff
@@ -6,244 +6,155 @@ import Private from '../private'
...
       overwrite = null
     }
 
-    if (typeof File !== 'undefined' && !(fileContent instanceof File)) {
-      fileContent = new Blob([fileContent])
-    }
+    fileContent = toBiteArray(fileContent)
 
-    if (fileContent.size > 2800000) {
+    if (getContentSize(fileContent) > 2800000) {
```
Comment (Valodya) [inline src/file/files.js]: please move this value `2800000` as constant to the top file

File: src/private.js [hunk 3]
```diff
@@ -1,40 +1,32 @@
...
-  setUIState: uiState => {
-    UIState = uiState
-  },
-
-  getUIState: () => {
-    return UIState
-  },
-
-  setDataToStore: (key, value) => {
+  setDataToStore: function(key, value) {
```
Comment (Valodya) [inline src/private.js]: here and for another methods use

```
setDataToStore(key, value) {
...
}
```

Comment (illyaV) [reply src/private.js]: done

File: src/file/files.js (hunk 2 above)
Comment (illyaV) [reply src/file/files.js]: done

File: src/file/files.js (hunk 1 above)
Comment (illyaV) [reply src/file/files.js]: done

**Pull request merged**